daemon/containerd/image_children.go

```diff
@@ -127,3 +127,79 @@ func isRootfsChildOf(child ocispec.RootFS, parent ocispec.RootFS) bool {
 
 	return true
 }
+
+// parents returns a slice of image IDs whose entire rootfs contents match,
+// in order, the childs first layers, excluding images with the exact same
+// rootfs.
+//
+// Called from image_delete.go to prune dangling parents.
+func (i *ImageService) parents(ctx context.Context, id image.ID) ([]imageWithRootfs, error) {
+	target, err := i.resolveDescriptor(ctx, id.String())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get child image")
+	}
+
+	cs := i.client.ContentStore()
+
+	allPlatforms, err := containerdimages.Platforms(ctx, cs, target)
+	if err != nil {
+		return nil, errdefs.System(errors.Wrap(err, "failed to list platforms supported by image"))
+	}
+
+	var childRootFS []ocispec.RootFS
+	for _, platform := range allPlatforms {
+		rootfs, err := platformRootfs(ctx, cs, target, platform)
+		if err != nil {
+			if !cerrdefs.IsNotFound(err) {
+				logrus.WithFields(logrus.Fields{
+					logrus.ErrorKey: err,
+					"image":         target.Digest,
+					"platform":      platform,
+				}).Warning("failed to get platform-specific rootfs")
+			}
+			continue
```

Comment:
@vvoland little sanity check as I'm not sure what other errors this could throw, do you think changing this could break anything?

Reply:
I think it's fine to return early in `parents` as the failure here could result in deleting an image that shouldn't be deleted. 
As for the `Children` - it's mostly used for `image ls` which probably shouldn't be errored out completely just because there was an issue with reading a single image/blob.
+1 on handling the context cancellation though for `Children` too though!